File: src/local_education_app/lib/models/slide/slide.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
class Slide {
  String id;
  String lessonId;
  String title;
  String content;
  String layout;
  int index;
  String thumbnailPath;
  String urlPath;
  bool isPublished;
  bool isTest;
  int minPoint;
  Slide({
    required this.id,
    required this.lessonId,
    required this.title,
    required this.content,
    required this.layout,
    required this.index,
    required this.thumbnailPath,
    required this.urlPath,
    required this.isPublished,
    required this.isTest,
    required this.minPoint,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'id': id,
      'lessonId': lessonId,
      'title': title,
      'content': content,
      'layout': layout,
      'index': index,
      'thumbnailPath': thumbnailPath,
      'urlPath': urlPath,
      'isPublished': isPublished,
      'isTest': isTest,
      'minPoint': minPoint,
    };
  }

  factory Slide.fromMap(Map<String, dynamic> map) {
    return Slide(
      id: map['id'] as String,
      lessonId: map['lessonId'] as String,
      title: map['title'] as String,
      content: map['content'] as String,
      layout: map['layout'] as String,
      index: map['index'] as int,
      thumbnailPath: map['thumbnailPath'] as String,
      urlPath: map['urlPath'] as String,
      isPublished: map['isPublished'] as bool,
      isTest: map['isTest'] as bool,
      minPoint: map['minPoint'] as int,
    );
  }
}

File: src/local-education-api/LocalEducation.Core/Entities/Slide.cs
using LocalEducation.Core.Contracts;

namespace LocalEducation.Core.Entities;

public class Slide : IEntity
{
    public Guid Id { get; set; }

    public DateTime CreatedDate { get; set; }

    public Guid LessonId { get; set; }

    public string Title { get; set; }

    public string Content { get; set; }

    public int Index { get; set; }

    public string Layout { get; set; }

    public string ThumbnailPath { get; set; }

    public string UrlPath { get; set; }

    public bool IsPublished { get; set; }

    public bool IsTest { get; set; }

    public int MinPoint { get; set; }

    // ======================================================
    // Navigation properties
    // ======================================================

    public Lesson Lesson { get; set; }

    public IList<Question> Questions { get; set; }

    public IList<ResultDetail> ResultDetails { get; set; }
}